### PR TITLE
Type definitions

### PR DESCRIPTION
**Pull request by Steams**
There is a very long list of currency codes and currency exponents for each country in the doc. If there is a specific subset of these you are interested I can pull them out into defined types

### CODE REVIEW

File: gateways/orbital/types.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+import "encoding/xml"
+
+type RequestType string
+
+const (
+	RequestTypeAuth    = "NewOrder"
+	RequestTypeCapture = "MarkForCapture"
+	RequestTypeVoid    = "Reversal"
+	RequestTypeRefund  = "Reversal"
```
Comment (nirajjayantbolt) [inline gateways/orbital/types.go:11]: No need for both

File: gateways/orbital/types.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	AVSResponseNotChecked             AVSResponseCode = "3"
+	AVSResponseSkipped4               AVSResponseCode = "4"
+	AVSResponseSkippedR               AVSResponseCode = "R"
+	AVSResponseMatch                  AVSResponseCode = "H"
+	AVSResponseNoMatch                AVSResponseCode = "G"
+	AVSResponseZipMatchAddressNoMatch AVSResponseCode = "A"
+	AVSResponseZipNoMatchAddressMatch AVSResponseCode = "F"
+)
+
+type Request struct {
```
Comment (nirajjayantbolt) [inline gateways/orbital/types.go:73]: these seem a little odd to me, why XML Name?

File: gateways/orbital/types.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+type RequestBody struct {
+	XMLName                   xml.Name
+	OrbitalConnectionUsername string        `xml:"OrbitalConnectionUsername"`
+	OrbitalConnectionPassword string        `xml:"OrbitalConnectionPassword"`
+	BIN                       BIN           `xml:"BIN"`
+	TerminalID                string        `xml:"TerminalID"` // usually 001, for PNS can be 001 - 999 but usually 001
+	IndustryType              IndustryType  `xml:"IndustryType,omitempty"`
+	MessageType               MessageType   `xml:"MessageType,omitempty"`
+	MerchantID                int           `xml:"MerchantID,omitempty"`
+	AccountNum                string        `xml:"AccountNum,omitempty"`
```
Comment (nirajjayantbolt) [inline gateways/orbital/types.go:92]: is this cc number?

File: gateways/orbital/types.go (hunk 2 above)
Comment (Steams) [reply gateways/orbital/types.go:73]: All requests can use the same structure and properties but the name of the actual tag when marshaled to XML has to change, xmlName lets you change it. EG :

an auth request looks like:
```
<Request> //This is the Request Struct
  <NewOrder>...</NewOrder> //this is a RequestBody struct with xmlName = "NewOrder"
</Request>
```

a void request looks like:
```
<Request> //This is the Request Struct
  <Reversal>...</Reversal> //this is a RequestBody struct with xmlName = "Reversal"
</Request>
```

Setting it looks like this :
`orbitalAuthRequest.Body.XMLName = xml.Name{Local: RequestTypeAuth}`

File: gateways/orbital/types.go (hunk 3 above)
Comment (Steams) [reply gateways/orbital/types.go:92]: Yes

File: gateways/orbital/types.go (hunk 2 above)
Comment (nirajjayantbolt) [reply gateways/orbital/types.go:73]: Thanks got it

File: gateways/orbital/types.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	RequestTypeCapture  = "MarkForCapture"
+	RequestTypeVoid     = "Reversal"
+)
+
+type MessageType string
+
+const (
+	MessageTypeAuth           MessageType = "A"
+	MessageTypeAuthAndCapture MessageType = "AC"
+	MessageTypeCapture        MessageType = "FC"
```
Comment (dbook13) [inline gateways/orbital/types.go:18]: Can we call this `MessageTypeForceAndCapture`? It seems like there is a slight difference between a Force and Capture and a normal capture, and I want to make sure they don't get confused here

Comment (Steams) [reply gateways/orbital/types.go:18]: Sure thing, just to be clear though, the Capture method i implemented actually uses the "MarkForCapture" request type, as opposed to a NewOrder request with the AuthAndCApture or ForceCapture message types. 
Similar to reversal vs refund earlier, is it that the method of capturing we're interested in should use the NewOrder request and not MarkForCapture ?

Comment (dbook13) [reply gateways/orbital/types.go:18]: We want to use the "MarkForCapture" request type for the Capture method like you've already implemented. I just wanted the naming of this message type to make it clear that it is not a secondary method of performing a transaction and actually does something different.

**Pull request merged**